File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        //The optimised approach is moore's voting algorithm
        int ele = nums[0];
        int count = 1;
        for(int i = 1; i<nums.length; i++){
            if(ele==nums[i]){
                count++;
            }else{
                count--;
            }
            if(count==0){
                ele = nums[i];
                count = 1;
            }
        }
        return ele;
    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        hash_map = {}
        for i in range(len(nums)):
            if nums[i] not in hash_map:
                hash_map[nums[i]] = 0
            else:
                hash_map[nums[i]] += 1
        return [i for i in hash_map if hash_map[i]==max(hash_map.values())][0]
            
            
        